src/AWS.Deploy.Orchestration/OptionSettingHandler.cs

```diff
@@ -48,15 +49,19 @@ public List<ValidationResult> RunOptionSettingValidators(Recommendation recommen
                     .Where(x => !x.IsValid)
                     .ToList());
 
-                if (settingValidatorFailedResults.Any())
+                if (optionSetting.Validation.InvalidValue == null)
                 {
-                    optionSetting.Validation.ValidationStatus = ValidationStatus.Invalid;
-                    optionSetting.Validation.ValidationMessage = string.Join(Environment.NewLine, settingValidatorFailedResults.Select(x => x.ValidationFailedMessage)).Trim();
-                }
-                else
-                {
-                    optionSetting.Validation.ValidationStatus = ValidationStatus.Valid;
-                    optionSetting.Validation.ValidationMessage = string.Empty;
+                    if (settingValidatorFailedResults.Any())
+                    {
+                        optionSetting.Validation.ValidationStatus = ValidationStatus.Invalid;
+                        optionSetting.Validation.ValidationMessage = string.Join(Environment.NewLine, settingValidatorFailedResults.Select(x => x.ValidationFailedMessage)).Trim();
+                        optionSetting.Validation.InvalidValue = optionSettingValue;
+                    }
+                    else
+                    {
+                        optionSetting.Validation.ValidationStatus = ValidationStatus.Valid;
+                        optionSetting.Validation.ValidationMessage = string.Empty;
```

Comment:
Line 52 isn't clear to me - when will be be here with `InvalidValue` already set?

----
If this is doing something like _"only update the validation status if it wasn't previously already invalid"_, I'd rather check the previous `ValidationStatus` instead of the presence of an `InvalidValue` - what if the reason it was invalid is because `null` is the invalid value?